### PR TITLE
Adding translations to source (update via contrib/translation-download.sh)

### PR DESCRIPTION
**Pull request by monotek**
packager.io is constantly fails building new packages because of the translation download, so i put the translations to the source.

There is a ne file contrib/translation-download.sh which downloads translations using a temporary sqlite db. This script would need to be run before a Zammad release to update all translations.

P.S: No need to merge. I'll put it in git.znuny.com if everybody is ok with it.

### CODE REVIEW

**Comment by martini** [comment]
If I got it correctly, you want to put the translation download files (.yml) into the git repo?

So you do not want to call contrib/localisation-download.sh via packager.io, only for the packages?

**Comment by monotek** [comment]
correct. imho this process needs to long in packager.io. 
imho thats the reason the builds are failing.
therefore it would be better to have this already in the sources.

**Comment by thorsteneckel** [comment]
I don't think that the Zammad source repo is the right place for the translations. This will lead to a lot of side efffects from pull requests to manual syncs etc... What concrete leads to the aborted processes? Can we maybe increase a timeout somewhere?

**Comment by monotek** [comment]
packager.io is kind of a blackbox for us.
Not exactly sure whats the problem.
The database dies with:

```
WARNING:  terminating connection because of crash of another server process
       DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
       HINT:  In a moment you should be able to reconnect to the database and repeat your command.
       rake aborted!
```

i'll try to switch to sqlite instead of postgresql. Maybe it helps.

If not, what to do? Separate (sub)repo for translations?

**Comment by martini** [comment]
> If not, what to do? Separate (sub)repo for translations?

I'll try 10 lines ruby today.

**Comment by monotek** [comment]
Switch to sqlite does not work. Postgresql is used anyway...

**Comment by monotek** [comment]
Any news on this topic?

Imho it would be better to be able to fetch this stuff via wget instead needing a running rails environment & database...

**Comment by martini** [comment]
@monotek here you go https://github.com/zammad/zammad/commit/f9c104093033b08a530d9ced36f70cdd2febb3ee it a ruby only (without rails) implementation.

**Comment by monotek** [comment]
Great! Now we have really nice & fast GitHub to packager.io to dockerhub buildchain :-)

**Comment by martini** [comment]
Package build also looks to be stable again!  🤓

**Comment by monotek** [comment]
On small issue. New package size is more than 40MB larger :-/

Before around 65MB:
https://packager.io/gh/zammad/zammad/build_runs/1428

Now around 107MB:
https://packager.io/gh/zammad/zammad/build_runs/1439

Maybe there could be done some cleanup in the after script?
I guess the translations are not 40 MB big?

**Comment by martini** [comment]
du -sh config/translations/
3.6M	config/translations/

No, only 3,6 MB. 

If I compare
https://packager.io/gh/zammad/zammad/build_runs/1326 -> (91.65 MiB)

and

https://packager.io/gh/zammad/zammad/build_runs/1439 -> (107.49 MiB)

seems to be fine for me.

If I install the cents 7 package I see the following:

[root@localhost ~]# du -sh /opt/zammad/
667M	/opt/zammad/
[root@localhost ~]# du -sh /opt/zammad/tmp/cache/assets/sprockets/v3.0/
423M	/opt/zammad/tmp/cache/assets/sprockets/v3.0/
[root@localhost ~]# du -sh /opt/zammad/vendor/
200M	/opt/zammad/vendor/
[root@localhost ~]# 

Main content is in:
 -> sprockets -> Precompiled assets -> is ok - precompile assets will build faster next time btw. contrib/packager.io/postinstall.sh is faster
 -> vendor -> no chance to make it smaller

-=> For me it seems that packager io has changed something in the build process.

Win: My cents 7 installation was up and running within 2 min. because of already Precompiled assets
 -=> So for me the big package looks better as the small one.

**Comment by monotek** [comment]
Not sure if its true that packager.io changed something. 
The bigger filesize showed up first time when the switch to the new download method was made.
I guess because you removed "rm -rf tmp/cache*" from the after script and thats a difference of 1,2GB uncompressed.

Has this dir something to do with the language files?
If not, is it realy needed?

**Comment by martini** [comment]
> Has this dir something to do with the language files?

No.

> If not, is it realy needed?

It speed up the package installation from about 7 min to 2 min :-o

**Comment by martini** [comment]
> The bigger filesize showed up first time when the switch to the new download method was made.

No it was "sometimes"  🤔 also before:
https://packager.io/gh/zammad/zammad/build_runs/1326 -> (91.65 MiB)

**Comment by monotek** [comment]
Yes. I just compared to the version before.

The question to me is: Do we need tmp/cache in the package or not?

**Comment by martini** [comment]
> The question to me is: Do we need tmp/cache in the package or not?

Yea. It speed up the package installation from about 7 min to 2 min :-o

File: config/locales.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  active: true
+- locale: vi
+  alias: ''
+  name: Vietnam (ViɆt Nam)
+  active: true
+- locale: zh-cn
+  alias: ''
+  name: Chinese (Sim.) (简体中文)
+  active: true
+- locale: zj-tw
```
Comment (NeverMin) [inline config/locales.yml]: Hi @martini , How to fix it, maybe it is `locale: zh-tw` ?

**Pull request closed without merging**